### DISCUSSION TITLE
Exception happening only with compiled program, it does not happen with the source.

### DISCUSSION THREAD

**Discussion by ronaldofas** [marcelotduarte/cx_Freeze#1008] [Q&A] [answered]
Guys I have a problem and I wanted to see if you can help me, I created an application that reads a quantity of SQL records (about 29000 records) makes a treatment and writes the records (the same 29000 with more columns) in another table. The application works normally when I run the source file directly, however I need to compile the file into an executable to distribute on the drive and I have already tried to compile with cx_freeze and pyinstaller. With both libraries the compilation works normally and the application also runs but at the time the program is writing data to the sql server it generates an exception and closes the program with the following error messages:

`Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 960, in _find_and_load
SystemError: <class '_frozen_importlib._ModuleLockManager'> returned a result with an error set

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "converter_w63x.py", line 54, in <module>
  File "src \ infrastructure \ repository \ RepositorioDiario.py", line 39, in grav_rel_diario
  File "src \ infrastructure \ repository \ RepositorioDiario.py", line 169, in __gravar_saldos_rel_diario
  File "<frozen importlib._bootstrap>", line 960, in _find_and_load
SystemError: <class '_frozen_importlib._ModuleLockManager'> returned a result with an error set
`

![image](https://user-images.githubusercontent.com/16448028/114242206-e3a3b900-9960-11eb-9b38-e372592227e6.png)

I have already researched a lot on the internet and found no satisfactory solution, this last error points to some recursion problem, however I do not use this feature in my code, I suspect that it is a problem in the executable generation libraries, but I am not sure.
 
Has anyone been through this, would you have any tips to give me about this problem?

**Comment by marcelotduarte** [comment, marked as answer]
What SQL libraries are you using?

**Reply by ronaldofas** [reply]
I'm using SQL server with pyodbc.

**Reply by ronaldofas** [reply]
Sorry for poor informations before, I know.

SQL librarie: `SQL Server Native Client 11.0`
pyODBC: `pyodbc==4.0.30`

I have reduced the range captured in database to 1000 and tested again and discover the error in insert in database like this:
```
Traceback (most recent call last): 
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load 
  File "<frozen importlib._bootstrap>", line 948, in _find_and_load_unlocked 
ModuleNotFoundError: No module named 'uuid' 

The above exception was the direct cause of the following exception: 
Traceback (most recent call last): 
  File "<frozen importlib._bootstrap>", line 960, in _find_and_load 
SystemError: <class '_frozen_importlib._ModuleLockManager'> returned a result with an error set 

```
I'm using python 3.6.0 because it is the version used in company and I can't upgrade.

**Reply by marcelotduarte** [reply]
With information I can help ;-)
Use this command line to build and report:
python setup.py build_exe --includes=uuid

If you're using includes in the setup.py then you should add to it because it has precedence over the command line, I think.
On success, I can put a hook in the next release.

> I'm using python 3.6.0 because it is the version used in company and I can't upgrade.

cx_freeze supports it, then no problem, but there are some rules in some companies that favor crackers.

**Reply by ronaldofas** [reply]
I've tried it now and obtained the error:

```Traceback (most recent call last): 
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load 
  File "<frozen importlib._bootstrap>", line 948, in _find_and_load_unlocked 
ModuleNotFoundError: No module named 'decimal' 

The above exception was the direct cause of the following exception: 

Traceback (most recent call last): 
  File "<frozen importlib._bootstrap>", line 960, in _find_and_load 
SystemError: <class '_frozen_importlib._ModuleLockManager'> returned a result with an error set ```

I've tried again and it worked, thanks for the help!

Ps: Cara, acredita que agora que eu vi que você é brasileiro!? Mantive a resposta em inglês para ajudar outras pessoas. Muito obrigado pelo seu maravilhoso trabalho com essa biblioteca, está sendo muito útil para mim!

**Reply by marcelotduarte** [reply]
ok, use  --includes=uuid,decimal
maybe you should add others, probably datetime,locale,hashlib

**Reply by ronaldofas** [reply]
Ok, thanks again!

Maybe can I add this modules in packages on the setup.py?

**Reply by marcelotduarte** [reply]
> Ps: Cara, acredita que agora que eu vi que você é brasileiro!? Mantive a resposta em inglês para ajudar outras pessoas. Muito obrigado pelo seu maravilhoso trabalho com essa biblioteca, está sendo muito útil para mim! 

Espero ter ajudado!

> Maybe can I add this modules in packages on the setup.py?

Yes. Check docs: https://cx-freeze.readthedocs.io/en/latest/distutils.html#build-exe

Tomorrow I'll release a new version and add hooks for pyodbc

**Reply by marcelotduarte** [reply]
cx_Freeze 6.6 has just been released and supports this as a hook.
